extension/lib/magento/customer/getDiff.js

```diff
@@ -6,21 +6,22 @@ const _ = {
 /**
  * @param {StepContext} context
  * @param {Object} input
- * @param {array} input.favoriteList - products to put
- * @param {array} input.magentoProducts - products currently on the magento fav list
+ * @param {Object[]} input.favoriteList - products to put
+ * @param {Object[]} input.magentoProducts - products currently on the magento fav list
  * @returns {Promise<{productIds: Object}>}
  */
 module.exports = async (context, input) => {
   const { favoriteList, magentoProducts } = input
   let favIds = []
   let existIds = []
 
-  favoriteList.map(item => {
+  for (let item of favoriteList) {
```

Comment:
not what I meant, this can be made really simple:
```
const favIds = favoriteList.map(item => item.id)
const existIds = magentoProducts.map(item => item.id)
```